pymc4/inference/sampling.py

```diff
@@ -174,7 +187,17 @@ def logpfn(*values, **kwargs):
         _, st = flow.evaluate_model_transformed(model, state=st)
         return st.collect_log_prob()
 
-    return logpfn, dict(state.all_unobserved_values)
+    @tf.function(autograph=False)
+    def deterministics_callback(*values, **kwargs):
+        if kwargs and values:
```

Comment:
So what you actually want to do is a second model pass. Right? I agree it is the only solution without tfp codebase change. Looks like an acceptable solution. I hope tf is smart enough to optimize the execution graph, can you please check the slowdown?

Reply:
I'll try to measure the slowdown and post here shortly

Reply:
I measured the slowdown [here](https://github.com/pymc-devs/pymc4/pull/161#issuecomment-555506333). It's around 2.5%